pallets/allocations/src/migrations.rs

```diff
@@ -0,0 +1,162 @@
+/*
+ * This file is part of the Nodle Chain distributed at https://github.com/NodleCode/chain
+ * Copyright (C) 2020-2022  Nodle International
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+pub mod v1 {
+	use crate::{Config, Pallet};
+	use frame_support::{
+		dispatch::GetStorageVersion,
+		pallet_prelude::PhantomData,
+		storage::migration::{have_storage_value, remove_storage_prefix},
+		traits::{Get, OnRuntimeUpgrade},
+		weights::Weight,
+	};
+
+	pub struct MigrateToBoundedOracles<T>(PhantomData<T>);
+	impl<T: Config> OnRuntimeUpgrade for MigrateToBoundedOracles<T> {
+		fn on_runtime_upgrade() -> Weight {
+			let current = Pallet::<T>::current_storage_version();
+			let onchain = Pallet::<T>::on_chain_storage_version();
+
+			log::info!(
+				"on_runtime_upgrade[{:#?}]=> Running migration with current storage version {:?} / onchain {:?}",
+				line!(),
+				current,
+				onchain
+			);
+
+			if current == 1 && onchain == 0 {
+				let pallet_prefix: &[u8] = b"Allocations";
+				let storage_item_prefix: &[u8] = b"Oracles";
+
+				if have_storage_value(pallet_prefix, storage_item_prefix, &[]) {
+					remove_storage_prefix(pallet_prefix, storage_item_prefix, &[]);
+
+					current.put::<Pallet<T>>();
```

Comment:
Can you please take an explicit approach similar to https://github.com/paritytech/substrate/blob/polkadot-v0.9.20/frame/staking/src/migrations.rs#L50 to write the new storage version on chain here?

Reply:
Migrated handling of `StorageVersion`, similar to shared substrate `staking` pallet.